flink-core/src/main/java/org/apache/flink/api/connector/source/SplitEnumeratorContext.java

```diff
@@ -126,6 +126,16 @@ default void assignSplit(SplitT split, int subtask) {
      */
     void signalNoMoreSplits(int subtask);
 
+    /**
+     * Signals a subtask that it will not receive split for current source, but it will receive
+     * split for next sources. A common scenario is HybridSource. This indicates that task not truly
+     * read finished.
+     *
+     * @param subtask The index of the operator's parallel subtask that shall be signaled it will
+     *     receive splits later.
+     */
+    default void signalIntermediateNoMoreSplits(int subtask) {}
```

Comment:
Yes. SplitEnumeratorContext is a Public api. i add it with default method to avoid impact. But it is indeed inappropriate here. Let me modify it. Thanks for reviewing.

Reply:
I think we can add an interface in `flink-core`, e.g. 
```
@Internal
interface SupportsIntermediateNoMoreSplits {
    void signalIntermediateNoMoreSplits(int subtask);
}
```

`SourceCoordinatorContext` should implements it. And `HybridSourceSplitEnumerator` should check that the realContext is an instance of `SupportsIntermediateNoMoreSplits`.